### DISCUSSION TITLE
Adding option to use ROSC (internal oscillator)

### DISCUSSION THREAD

**Discussion by Ing-Dom** [earlephilhower/arduino-pico#243] [Ideas]
Hi,

I' like to discuss if it is possible to add an option to configure the RP2040 to run on the internal ROSC.
The ROSC has some advantages that can be usefull (like low power).

At least for the generic board it would be great, but also for the other boards it would be cool, because then you can test this feature with a standard board...

So what do you think?
Maybe you can also give me a hint where I can find XOSC configuration in the core, so I can try around a little myself..

**Comment by earlephilhower** [comment]
In `main.cpp` is where you'd want to do this:
https://github.com/earlephilhower/arduino-pico/blob/47a4d9f8033c6d96c858a3ed9ada46391f7c8bbf/cores/rp2040/main.cpp#L56-L59

You'd probably want to substitute the set_sys_clock call with whatever magic is needed to make the HW use the ROSC.  You can use the `pico-sdk`calls freely to make that happen.  I'd be worried about serial baud rate drift and USB errors, though, but if you're running embedded then it's probably not an issue.


If you get it working, then we can add a menu option to the over/underclocking to enable it.

Good luck!
